src/app-layer-parser.c

```diff
@@ -919,9 +919,30 @@ void AppLayerParserTransactionsCleanup(Flow *f, const uint8_t pkt_dir)
     SCLogDebug("start min %"PRIu64, min);
     bool skipped = false;
     // const bool support_files = AppLayerParserSupportsFiles(f->proto, f->alproto);
+    uint32_t tofree_nb = 0;
```

Comment:
would like this to live in a (local) util function to keep `AppLayerParserTransactionsCleanup` a bit cleaner

Reply:
Ok with some `AppLayerPrepareCleanupTxList`